// table/dist/test/table.js
"use strict";
/* eslint-disable max-nested-callbacks */
Object.defineProperty(exports, "__esModule", { value: true });
const chai_1 = require("chai");
const src_1 = require("../src");
const utils_1 = require("./utils");
const data = [
    ['Lorem ipsum', 'dolor sit'],
    ['amet', 'consectetur'],
    ['adipiscing', 'elit'],
];
const basicConfig = {
    columnDefault: {
        width: 5,
    },
};
describe('drawTable', () => {
    describe('drawHorizontalLine', () => {
        context('only draw top and bottom borders', () => {
            it('draws proper borders', () => {
                const config = {
                    ...basicConfig,
                    drawHorizontalLine: (index, size) => {
                        return index === 0 || index === size;
                    },
                };
                (0, chai_1.expect)((0, src_1.table)(data, config)).to.be.deep.equal(`
╔═══════╤═══════╗
║ Lorem │ dolor ║
║ ipsum │ sit   ║
║ amet  │ conse ║
║       │ ctetu ║
║       │ r     ║
║ adipi │ elit  ║
║ scing │       ║
╚═══════╧═══════╝
`.trimLeft());
            });
        });
        context('only draw inner borders', () => {
            it('draws proper borders', () => {
                const config = {
                    ...basicConfig,
                    drawHorizontalLine: (index, size) => {
                        return index > 0 && index < size;
                    },
                };
                (0, chai_1.expect)((0, src_1.table)(data, config)).to.be.deep.equal(`
║ Lorem │ dolor ║
║ ipsum │ sit   ║
╟───────┼───────╢
║ amet  │ conse ║
║       │ ctetu ║
║       │ r     ║
╟───────┼───────╢
║ adipi │ elit  ║
║ scing │       ║
`.trimLeft());
            });
        });
        context('only draw top and next-to-last borders', () => {
            it('draws proper borders', () => {
                const config = {
                    ...basicConfig,
                    drawHorizontalLine: (index, size) => {
                        return index === 0 || index === size - 1;
                    },
                };
                (0, chai_1.expect)((0, src_1.table)(data, config)).to.be.deep.equal(`
╔═══════╤═══════╗
║ Lorem │ dolor ║
║ ipsum │ sit   ║
║ amet  │ conse ║
║       │ ctetu ║
║       │ r     ║
╟───────┼───────╢
║ adipi │ elit  ║
║ scing │       ║
`.trimLeft());
            });
        });
    });
    context('header', () => {
        it('draw properly', () => {
            const config = {
                ...basicConfig,
                header: {
                    content: 'This is the long long header',
                },
            };
            (0, chai_1.expect)((0, src_1.table)(data, config)).to.be.deep.equal(`
╔═══════════════╗
║ This is the l ║
║ ong long head ║
║      er       ║
╟───────┬───────╢
║ Lorem │ dolor ║
║ ipsum │ sit   ║
╟───────┼───────╢
║ amet  │ conse ║
║       │ ctetu ║
║       │ r     ║
╟───────┼───────╢
║ adipi │ elit  ║
║ scing │       ║
╚═══════╧═══════╝
`.trimLeft());
        });
    });
    context('vertical alignment', () => {
        // eslint-disable-next-line @typescript-eslint/no-shadow
        const data = [
            ['Lorem ipsum dolor sit amet, consectetur adipiscing elit',
                'Phasellus pulvinar nibh sed',
                'Phasellus pulvinar nibh sed',
                'Phasellus pulvinar nibh sed'],
        ];
        it('works properly', () => {
            const result = (0, src_1.table)(data, {
                columnDefault: {
                    width: 10,
                },
                columns: [
                    {},
                    { verticalAlignment: 'top' },
                    { verticalAlignment: 'middle' },
                    { verticalAlignment: 'bottom' }
                ],
            });
            (0, utils_1.expectTable)(result, `
╔════════════╤════════════╤════════════╤════════════╗
║ Lorem ipsu │ Phasellus  │            │            ║
║ m dolor si │ pulvinar n │ Phasellus  │            ║
║ t amet, co │ ibh sed    │ pulvinar n │            ║
║ nsectetur  │            │ ibh sed    │ Phasellus  ║
║ adipiscing │            │            │ pulvinar n ║
║ elit       │            │            │ ibh sed    ║
╚════════════╧════════════╧════════════╧════════════╝`);
        });
        it('works with horizontal alignment', () => {
            const result = (0, src_1.table)(data, {
                columnDefault: {
                    width: 8,
                    wrapWord: true,
                },
                columns: [
                    {},
                    { alignment: 'center',
                        verticalAlignment: 'top' },
                    { alignment: 'right',
                        verticalAlignment: 'middle' },
                    { alignment: 'left',
                        verticalAlignment: 'bottom' }
                ],
            });
            (0, utils_1.expectTable)(result, `
╔══════════╤══════════╤══════════╤══════════╗
║ Lorem    │ Phasellu │          │          ║
║ ipsum    │    s     │          │          ║
║ dolor    │ pulvinar │ Phasellu │          ║
║ sit      │ nibh sed │        s │          ║
║ amet,    │          │ pulvinar │          ║
║ consecte │          │ nibh sed │ Phasellu ║
║ tur      │          │          │ s        ║
║ adipisci │          │          │ pulvinar ║
║ ng elit  │          │          │ nibh sed ║
╚══════════╧══════════╧══════════╧══════════╝`);
        });
    });
});
