// table/dist/test/validateConfig.js
"use strict";
/* eslint-disable max-nested-callbacks */
Object.defineProperty(exports, "__esModule", { value: true });
const chai_1 = require("chai");
const validateConfig_1 = require("../src/validateConfig");
describe('validateConfig', () => {
    context('given invalid config', () => {
        it('throws an error', () => {
            (0, chai_1.expect)(() => {
                (0, validateConfig_1.validateConfig)('config.json', { x: 1 });
            }).to.be.throw(Error, 'Invalid config.');
        });
    });
    context('given valid config', () => {
        it('does not throw an error', () => {
            (0, chai_1.expect)(() => {
                (0, validateConfig_1.validateConfig)('config.json', {
                    columnDefault: {
                        width: 50,
                    },
                    columns: {
                        0: {
                            alignment: 'left',
                            width: 10,
                        },
                    },
                    drawHorizontalLine: () => {
                        return false;
                    },
                    singleLine: true,
                });
            }).to.be.not.throw();
        });
    });
});
